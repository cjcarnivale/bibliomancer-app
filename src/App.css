.hero{
  text-align: center; 
  line-height: 1.4em; 
}

.description{
  text-align: left; 
}

.container{
  padding-left: 10vw;
  padding-right: 10vw;
  padding-bottom: 10vw;  
  font-size: 1.4em;
}

.selection img {
  width: 40vw; 
  height: 40vw; 
  margin: 0 auto;
  display: block;  
}

.description {
  line-height: 1.4em;
  font-size: 1.1em;  
}

.controls{
  display: flex; 
  justify-content: center; 
}

body {
  background-color: #E3DDA8
}

button{
  font-size: 25px;
  border-radius: 10px; 
  padding: 5px; 
}

#new-book{
  background-color: #FF7E40; 
}

select{
  font-size: 25px;
  background-color: #B0924E 
}

a{
  text-decoration: none; 
  color: black; 
}

.demo-button-container{
  display: flex;
  justify-content: center; 
}

h1, h2, h3, h4{
  font-family: 'Playfair Display', serif;
}

p{
  font-family: 'Fauna One', serif;
}

#read-this{
  color: white; 
  background-color: #097E40;
}

.demo-button{
  background-color: #64C3D9
}

#read-it{
  background-color: #097E40;
}

#different-selection{
  background-color: #FFBD2F 
}

.selection-buttons{
  display: flex;
  justify-content: center; 
}

.read-this{
  margin-top: 20px; 
  display: flex;
  justify-content: center; 
}

.read img{
  width: 40vw; 
  height: 40vw;
  margin-left: 10px;  
}


@media only screen and (min-width: 600px){
  .selection img {
    width: 20vw; 
    height: 20vw; 
    margin: 0 auto;
    display: block;  
  }
  .read img{
    width: 10vw; 
    height: 10vw;
    margin-left: 10px;  
  }
}