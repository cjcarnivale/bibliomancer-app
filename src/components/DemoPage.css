.select-button{
  text-align: center
}
