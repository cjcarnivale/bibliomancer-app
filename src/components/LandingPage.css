h1.hero{
  text-align: center;
  font-size: 48px; 
}


.description{
  width: 800px; 
  margin: 0 auto; 
  line-height: 32px; 
  font-size: 20px; 
  margin-bottom: 50px; 
}

#book-cover{
  display: block;
  margin-left: auto;
  margin-right: auto; 
  margin-top: 50px; 
  margin-bottom:  25px;  
  width: 8%;
  
}

.book-details{
  width: 600px; 
  margin: 0 auto;
  list-style-type: none;  
}

.button-container{
  margin-top: 50px; 
  text-align: center; 
}

.demo{
  text-align: center; 
}

.demo-button{
  font-size: 20px; 
}
