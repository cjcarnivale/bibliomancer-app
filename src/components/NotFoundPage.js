import React, { Component } from 'react'; 

class NotFoundPage extends Component {
  render(){
    return(
      <div>
        <p>
          Oops! Page does not exist.
        </p>
      </div>
    ) 
  }
}

export default NotFoundPage; 