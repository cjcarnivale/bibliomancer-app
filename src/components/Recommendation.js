import React, { Component } from 'react';
import Context from '../Context';
import SelectButton from './SelectButton'

class Recommendation extends Component{
  static contextType = Context; 
    
  constructor(props){
      super(props);
      this.state = {
        reading: false
      }
    }

  handleReading = () => {
    this.setState({
      reading: !this.state.reading
    })
  }

  render() {
    const [preview] = this.context.current;
    if (preview === undefined){
      return (
        <div className="selection empty">
          <h3>
            No Remaining Book in {this.context.genre} Genre.
          </h3>
        </div>
      );
    } else {
      return (
        <div className="selection">
          <h3 id="title">{preview.title}</h3>
          <h4>by {preview.author}</h4>
          <img src={preview.image} alt="book cover"/>
          <br />
          <p className="description">{preview.description}</p>
          <div>
          {
          (this.props.showButton)
            ?
            (!this.state.reading)
              ?
              <div>
                <SelectButton />
                <div className="read-this">
                  <button id="read-this"onClick={this.handleReading}>
                    Read This
                  </button>
                </div>
              </div>
              :
              <div className="selection-buttons">
                <button id="different-selection" onClick={this.handleReading}>
                  Different Selection
                </button>
                <button id="read-it" onClick={() => {this.context.readIt(preview.id); this.handleReading()}}>
                  Read It!
                </button>
              </div>
            :
            <div></div>
          }
          </div>
        </div>
      );
    }
  }

}

export default Recommendation