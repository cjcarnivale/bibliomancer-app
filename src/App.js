import React, { Component } from 'react';
import { Route, Switch }from 'react-router-dom';
import LandingPage from './components/LandingPage';
import DemoPage from './components/DemoPage'; 
import NotFoundPage from './components/NotFoundPage'; 
import CommentsPage from './components/CommentsPage';
import './App.css';
class App extends Component {
  
  render(){

    return (
      <main className='App'>
          <Switch>
            <Route exact path='/' component={LandingPage}/>
            <Route path='/DemoPage' component={DemoPage}/>
            <Route path='/CommentsPage/:bookId' component={CommentsPage}/>
            <Route component={NotFoundPage}/>
          </Switch>
      </main>
      
    );
  }
}
export default App;