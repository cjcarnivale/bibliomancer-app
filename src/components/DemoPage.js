import React, { Component } from 'react'; 
import Recommendation from './Recommendation'
import Context from '../Context';

class DemoPage extends Component {

  static contextType = Context; 

  renderRead(){
    const readBooks = this.context.read.map((val, i ) => (
      <img key={i} src={val.image} alt="book cover" />
    ));
    return (
      <div>
        <h3>Read</h3>
        <div className="read">{readBooks}</div>
      </div>
    );
  }

  render(){
    return (
      <div className="container">
        <h1>Try It Out!</h1>
        <div className="current">
          <Recommendation showButton={true}/>
          <hr />
          {this.renderRead()}
        </div>
      </div>  
    ); 
  }
}

export default DemoPage; 