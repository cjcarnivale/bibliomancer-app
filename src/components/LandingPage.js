import React, { Component } from 'react'; 
import Recommendation from './GetRecommendation'
class LandingPage extends Component {
  constructor(props) {
    super(props); 
    this.state = {
      recommendation: {},
      genre:'fiction'
    };
  }

  setGenre = (event) => {
    this.setState({
      genre: event.target.value 
    })
  }

  getRecommendation = () => {
    fetch(`http://localhost:8000/api/recommendation?genre=${this.state.genre}`)
    .then(response => response.json())
    .then(data => {
      console.log(data)
      if(!this.state.recommendation.id || this.state.recommendation.id === data.length){
        const recommendation = data[0]; 
        this.setState({
          recommendation
        })
      }
      else{
        const recommendation = data[this.state.recommendation.id];
        this.setState({
          recommendation 
        })
      }
    })
  } 

  render(){
    return(
      <div>
        <div className="hero">
          <h1>Bibliomancer</h1>
          <p className="description">
            Are you ever stuck knowing that you want to read SOMETHING, but can't decide on what?  Do your interests include
            Fiction, Science Fiction/Fantasy, Classic Literature, or Self-Help?  If so, this is the place for you.  Bibliomancer 
            provides recommendations for you to take the guesswork out of what to read next.  You will also be able to track the books
            you have read and when finished with one, receive another recommendation.  Start reading again, today!
          </p>
        </div>
        <div className="recommendation">
          <Recommendation 
            getRecommendation={this.getRecommendation} recommendation={this.state.recommendation}
            setGenre={this.setGenre}/>
          <button type="button" onClick={this.getRecommendation}>Get New Recommendation</button>
          <select onChange={(event) => this.setGenre(event)}>
            <option value="fiction">Fiction</option>
            <option value="fantasy">Science Fiction/Fantasy</option>
            <option value='literature'>Literature</option>
            <option value='help'>Self-Help</option>
          </select>
        </div>
        <div className="feature-one">
            Get recommendations based on genre with screenshot
        </div>
        <div className="feature-two">
            Track books completed and badges with screenshot
        </div>
      </div>
    ) 
  }
}

export default LandingPage; 