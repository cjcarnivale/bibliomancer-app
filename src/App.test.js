import React from 'react';
import { StaticRouter } from 'react-router-dom'
import App from './App';
import NotFoundPage from "./components/NotFoundPage"
import renderer from 'react-test-renderer'; 
import LandingPage from './components/LandingPage';
import { BooksProvider } from './Context';
import DemoPage from './components/DemoPage';
import Recommendation from './components/Recommendation';
import SelectButton from './components/SelectButton';

const values = {
  allBooks: [
    {
      title: 'Atlas Shrugged', 
      author: 'Ayn Rand', 
      description: `Peopled by larger-than-life heroes and villains, charged with towering questions of good and evil, 
                    Atlas Shrugged is Ayn Rand’s magnum opus: a philosophical revolution told in the form of an action thriller—nominated 
                    as one of America’s best-loved novels by PBS’s The Great American Read.`,
      image: 'https://images-na.ssl-images-amazon.com/images/I/51TyZI03%2B9L._SX303_BO1,204,203,200_.jpg',
      genre: 'Fiction'
    },
    {
      title: 'Slaughterhouse-Five', 
      author: 'Kurt Vonnegut',
      description: `An American classic and one of the world’s great antiwar books. Centering on the infamous 
                    firebombing of Dresden, Billy Pilgrim’s odyssey through time reflects the mythic journey of our own fractured lives 
                    as we search for meaning in what we fear most.`,
      image: 'https://images-na.ssl-images-amazon.com/images/I/41tyM7I9EcL._SX328_BO1,204,203,200_.jpg',
      genre: 'Fiction'
    } 
  ],
  genres: {Fiction: []},
  genre: 'Fiction',
  current: [  
    {
    title: 'Atlas Shrugged', 
    author: 'Ayn Rand', 
    description: `Peopled by larger-than-life heroes and villains, charged with towering questions of good and evil, 
                  Atlas Shrugged is Ayn Rand’s magnum opus: a philosophical revolution told in the form of an action thriller—nominated 
                  as one of America’s best-loved novels by PBS’s The Great American Read.`,
    image: 'https://images-na.ssl-images-amazon.com/images/I/51TyZI03%2B9L._SX303_BO1,204,203,200_.jpg',
    genre: 'Fiction'
    },
  ],
  selectGenre: jest.fn(),
  readIt: jest.fn(),
  next: jest.fn(),
  read: {
    title: 'Slaughterhouse-Five', 
    author: 'Kurt Vonnegut',
    description: `An American classic and one of the world’s great antiwar books. Centering on the infamous 
                  firebombing of Dresden, Billy Pilgrim’s odyssey through time reflects the mythic journey of our own fractured lives 
                  as we search for meaning in what we fear most.`,
    image: 'https://images-na.ssl-images-amazon.com/images/I/41tyM7I9EcL._SX328_BO1,204,203,200_.jpg',
    genre: 'Fiction'
    }
  }

describe('Render all components', () => {
  it('renders App as expected', () => {
    const tree = renderer 
    .create(<StaticRouter><BooksProvider value={values}><App /></BooksProvider></StaticRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
  })
  it('renders Landing Page as expected', () => {
    const tree = renderer 
    .create(<StaticRouter><BooksProvider value={values}><LandingPage /></BooksProvider></StaticRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
  })
  it('renders Demo Page as expected', () => {
    const tree = renderer 
    .create(<StaticRouter><BooksProvider value={values}><DemoPage /></BooksProvider></StaticRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
  })
  it('renders Not Found Page as expected', () => {
    const tree = renderer 
    .create(<NotFoundPage />).toJSON();
    expect(tree).toMatchSnapshot(); 
  })
  it('renders Recommendation as expected', () => {
    const tree = renderer 
    .create(<StaticRouter><BooksProvider value={values}><Recommendation /></BooksProvider></StaticRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
  })
  it('renders SelectButton as expected', () => {
    const tree = renderer 
    .create(<StaticRouter><BooksProvider value={values}><SelectButton /></BooksProvider></StaticRouter>).toJSON();
    expect(tree).toMatchSnapshot(); 
  })
})




