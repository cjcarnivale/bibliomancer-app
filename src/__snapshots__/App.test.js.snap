// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render all components renders App as expected 1`] = `
<main
  className="App"
>
  <div
    className="container"
  >
    <div
      className="hero"
    >
      <h1>
        Bibliomancer
      </h1>
      <div
        className="description"
      >
        <p>
          Are you ever stuck knowing that you want to read SOMETHING, but can't decide on what? 
        </p>
        <p>
          Do your interests include Fiction, Science Fiction/Fantasy, Classic Literature, or Self-Help? If so, this is the place for you.
        </p>
        <p>
          Bibliomancer provides recommendations for you to take the guesswork out of what to read next. You will also be able to track the books you have read and when finished with one, receive another recommendation.
        </p>
        <p>
          Start reading again, today!
        </p>
      </div>
    </div>
    <div
      className="recommendation"
    >
      <h2>
        Take a Look at Our Recommendations
      </h2>
      <hr />
      <div
        className="selection empty"
      >
        <h3>
          No Remaining Book in 
          Fiction
           Genre.
        </h3>
      </div>
      <div
        className="controls"
      >
        <select
          onChange={[Function]}
        />
        <button
          id="new-book"
          onClick={[Function]}
        >
          <a
            href="#title"
          >
            New Book
          </a>
        </button>
      </div>
    </div>
    <div>
      <h2>
        Choose a book to read and track what you have read in our interactive demo!
      </h2>
      <div
        className="demo-button-container"
      >
        <a
          className="demo-page-link"
          href="/DemoPage"
          onClick={[Function]}
        >
          <button
            className="demo-button"
            type="button"
          >
            Interactive Demo
          </button>
        </a>
      </div>
    </div>
  </div>
</main>
`;

exports[`Render all components renders Demo Page as expected 1`] = `
<div
  className="container"
>
  <div
    className="current"
  >
    <div
      className="selection empty"
    >
      <h3>
        No Remaining Book in 
        Fiction
         Genre.
      </h3>
    </div>
    <hr />
    <div>
      <h3>
        Read
      </h3>
      <div
        className="read"
      />
    </div>
  </div>
</div>
`;

exports[`Render all components renders Landing Page as expected 1`] = `
<div
  className="container"
>
  <div
    className="hero"
  >
    <h1>
      Bibliomancer
    </h1>
    <div
      className="description"
    >
      <p>
        Are you ever stuck knowing that you want to read SOMETHING, but can't decide on what? 
      </p>
      <p>
        Do your interests include Fiction, Science Fiction/Fantasy, Classic Literature, or Self-Help? If so, this is the place for you.
      </p>
      <p>
        Bibliomancer provides recommendations for you to take the guesswork out of what to read next. You will also be able to track the books you have read and when finished with one, receive another recommendation.
      </p>
      <p>
        Start reading again, today!
      </p>
    </div>
  </div>
  <div
    className="recommendation"
  >
    <h2>
      Take a Look at Our Recommendations
    </h2>
    <hr />
    <div
      className="selection empty"
    >
      <h3>
        No Remaining Book in 
        Fiction
         Genre.
      </h3>
    </div>
    <div
      className="controls"
    >
      <select
        onChange={[Function]}
      />
      <button
        id="new-book"
        onClick={[Function]}
      >
        <a
          href="#title"
        >
          New Book
        </a>
      </button>
    </div>
  </div>
  <div>
    <h2>
      Choose a book to read and track what you have read in our interactive demo!
    </h2>
    <div
      className="demo-button-container"
    >
      <a
        className="demo-page-link"
        href="/DemoPage"
        onClick={[Function]}
      >
        <button
          className="demo-button"
          type="button"
        >
          Interactive Demo
        </button>
      </a>
    </div>
  </div>
</div>
`;

exports[`Render all components renders Not Found Page as expected 1`] = `
<div>
  <p>
    Oops! Page does not exist.
  </p>
</div>
`;

exports[`Render all components renders Recommendation as expected 1`] = `
<div
  className="selection empty"
>
  <h3>
    No Remaining Book in 
    Fiction
     Genre.
  </h3>
</div>
`;

exports[`Render all components renders SelectButton as expected 1`] = `
<div
  className="controls"
>
  <select
    onChange={[Function]}
  />
  <button
    id="new-book"
    onClick={[Function]}
  >
    <a
      href="#title"
    >
      New Book
    </a>
  </button>
</div>
`;
